tower-sessions-core/src/session.rs

```diff
@@ -948,3 +951,57 @@ pub enum Expiry {
     /// [`set_expiry`](Session::set_expiry).
     AtDateTime(OffsetDateTime),
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    mod session {
+        use super::*;
+        use crate::session_store::MockSessionStore;
```

Comment:
Is there a reason to not use `MemoryStore` here? One of its primary reasons for existing is testing so that mocks can be avoided.

Reply:
I wanted a mocked store that I could explicitly set returns for. I don't want to depend on semantics of MemoryStore.

Reply:
Also, I think it's important not to have a circular dependency for unit tests.

MemoryStore is more appropriate for integration testing.